src/java/org/apache/cassandra/auth/FunctionResource.java

```diff
@@ -176,8 +177,10 @@ public static FunctionResource fromName(String name)
         if (parts.length == 2)
             return keyspace(parts[1]);
 
+        if (!name.matches("^.+\\[.*\\]$"))
+            throw new IllegalArgumentException(String.format("%s is not a valid function resource name. It must end with \"[]\"", name));
         String[] nameAndArgs = StringUtils.split(parts[2], "[|]");
-        return function(parts[1], nameAndArgs[0], argsListFromString(nameAndArgs[1]));
+        return function(parts[1], nameAndArgs[0], nameAndArgs.length > 1 ? argsListFromString(nameAndArgs[1]) : Collections.emptyList());
```

Comment:
I'm not sure this check make sense now because we added the same logical check by the pattern match. It guarantees that we won't receive the name without square brackets, so `nameAndArgs.length > 1 ` is always true.

Reply:
Hi @azotcsit when you split consecutive separators are considered a single one and are not returned as splits. So  the string `func[]` is actually split the array `[func]` meaning a single element array hence we need to keep that logic. Makes sense?

Reply:
Correct, you're right! Sorry for the confusion.